clients/src/test/java/org/apache/kafka/common/record/DefaultRecordBatchTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.record;
+
+import org.apache.kafka.common.utils.Utils;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+public class DefaultRecordBatchTest {
```

Comment:
Hmm, shouldn't we have more tests here? Looking at the tests for the legacy records, some ideas:

1. Tests like `SimpleRecordTest.*isValid*`, but for `DefaultRecordBatch`.
2. `LegacyRecordTest.testChecksum` would be useful to have as well.
3. `SimpleRecord.buildEosRecord` should be renamed and moved here.
4. Some tests involving compression? Or is the idea that `MemoryRecordsTest` and `MemoryRecordsBuilderTest` cover those? It may be worth adding some javadoc to the test classes to give an idea of the what they're intended to test.